tests/unit/auth/v2/queryAuthCheck.js

```diff
@@ -11,18 +12,111 @@ describe('v2: queryAuthCheck', () => {
         { token: undefined, error: false },
         { token: 'invalid-token', error: true },
         { token: 'a'.repeat(128), error: false },
-    ].forEach(test => it(`test with token(${test.token})`, () => {
-        const request = { method: 'GET' };
+    ].forEach(test => {
+        it(`test with token(${test.token})`, () => {
+            const request = { method: 'GET' };
+            const data = {
+                SecurityToken: test.token,
+            };
+            const res = queryAuthCheck(request, log, data);
+            if (test.error) {
+                assert.notStrictEqual(res.err, undefined);
+                assert.strictEqual(res.err.InvalidToken, true);
+            } else {
+                assert.notStrictEqual(res.err, undefined);
+                assert.strictEqual(res.err.MissingSecurityHeader, true);
+            }
+        });
+    });
+});
+
+describe('v2: queryAuthCheck', () => {
+    let clock;
+
+    beforeEach(() => {
+        clock = sinon.useFakeTimers();
+    });
+    afterEach(() => {
+        process.env.PRE_SIGN_URL_EXPIRY = 604800000;
+        clock.restore();
+    });
+    it('URL should not expire before 7 days with default expiry', () => {
+        const currentTime = Date.now() / 1000;
+        const expires = currentTime + 604799;  // in seconds
+        const mockRequest = {
+            method: 'GET',
+            url: 'mockurl',
+            query: {
+                Expires: expires,
+            },
+            headers: {
+                'Content-MD5': 'c',
+            },
+        };
         const data = {
-            SecurityToken: test.token,
+            Expires: expires,
+            AWSAccessKeyId: 'keyId',
+            Signature: 'sign',
         };
+        const res = queryAuthCheck(mockRequest, log, data);
+        assert.notStrictEqual(res.err.AccessDenied, true);
+        assert.notStrictEqual(res.err.RequestTimeTooSkewed, true);
+    });
+    it('URL should expire after 7 days with default expiry', () => {
+        clock.tick(604800000);  // take time 604800000ms (7 days) ahead
+        const currentTime = Date.now();
```

Comment:
This might sound kinda lame or cheesy but to call a "currentTime" after setting the time to 7 days ahead makes it a little confusing to me. Maybe like "expiredTime" or just testTime to be more generic for all tests.

Reply:
The idea is to emphasize that the URL expires or remains valid at the current time after 7 days, so "expiredTime" could be misleading, but using a generic testTime for all the tests is a good idea.